app/src/androidTest/java/ch/epfl/sdp/drone3d/ui/mission/SaveMappingMissionActivityTest.kt

```diff
@@ -124,136 +166,126 @@ class SaveMappingMissionActivityTest {
         Assert.assertEquals("ch.epfl.sdp.drone3d", appContext.packageName)
     }
 
+    init {
+        `when`(locationService.isLocationEnabled()).thenReturn(false)
+        `when`(weatherService.getWeatherReport(AREA[0])).thenReturn(
+            MutableLiveData(
+                GOOD_WEATHER_REPORT
+            )
+        )
+    }
+
 
     @Test
     fun saveMappingMissionToPrivateCallStore() {
         activityRule.scenario.recreate()
 
-        val expectedMappingMission = MappingMission(
-            "Unnamed mission"
-        )
-
         onView(withId(R.id.privateCheckBox)).perform(click())
         onView(withId(R.id.saveButton)).perform(click())
 
-        assertThat(
-            activityRule.scenario.state,
-            equalTo(Lifecycle.State.DESTROYED)
+        sleep(1000)
+
+        Intents.intended(
+            IntentMatchers.hasComponent(ComponentNameMatchers.hasClassName(ItineraryShowActivity::class.java.name))
         )
 
-        verify(mappingMissionDao, times(1)).storeMappingMission(USER_UID, expectedMappingMission)
+        val intents = Intents.getIntents()
+        assert(intents.any { it.hasExtra(MissionViewAdapter.FLIGHTHEIGHT_INTENT_PATH) })
+        assert(intents.any { it.hasExtra(MissionViewAdapter.STRATEGY_INTENT_PATH) })
+        assert(intents.any { it.hasExtra(MissionViewAdapter.AREA_INTENT_PATH) })
+
+        verify(mappingMissionDao, times(1)).storeMappingMission(USER_UID, UNNAMED_MISSION)
     }
 
     @Test
     fun saveMappingMissionToShareCallShare() {
         activityRule.scenario.recreate()
 
-        val expectedMappingMission = MappingMission("Unnamed mission")
-
         onView(withId(R.id.sharedCheckBox)).perform(click())
         onView(withId(R.id.saveButton)).perform(click())
 
-        assertThat(
-            activityRule.scenario.state,
-            equalTo(Lifecycle.State.DESTROYED)
+        sleep(1000)
```

Comment:
Is the sleep really necessary ?

Reply:
In ItineraryShowActivity, the mission is generated on creation of the activity, the sleep is here to let the time to init the map and build the mission